src/keys.test.ts

```diff
@@ -111,6 +112,25 @@ describe("typing", () => {
       expectTypeOf(result).toEqualTypeOf<[]>();
     });
 
+    test("unions of records", () => {
+      const dataFirst = keys(
+        {} as Record<number, string> | Record<string, string>,
```

Comment:
You should use `unknown` for the values type, to make it even more general.
```suggestion
        {} as Record<number, unknown> | Record<string, unknown>,
```